docs/blog/2025-01-20-linking-ts-in-monorepos.md

```diff
@@ -0,0 +1,666 @@
+   │     │  ├─ index.js.map
+   │     │  └─ tsconfig.tsbuildinfo
+   │     ├─ package.json
+   │     ├─ src
+   │     │  └─ index.ts
+   │     └─ tsconfig.json
+   ├─ tsconfig.base.json
+   └─ tsconfig.json
+```
+
+To make the compiled `lib-a` usable for other packages, we need to update its `package.json` to point to the compiled artifacts in `dist`. Here's the updated version:
+
+```json {% fileName="packages/lib-a/package.json" %}
+{
+  "name": "@ts-monorepo-linking/lib-a",
+  "version": "0.0.0",
+  "private": true,
+  "type": "module",
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "import": "./dist/index.js",
+      "default": "./dist/index.js"
+    },
+    "./package.json": "./package.json"
+  },
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "module": "./dist/index.js"
+}
+```
+
+### Observations
+
+**Dependency resolution:**  
+Precompiling dependent packages allows the application bundler to rely on prebuilt outputs, avoiding the need to compile package dependencies during application bundling. A [task pipeline](/concepts/task-pipeline-configuration) ensures that packages are compiled beforehand, streamlining the workflow.
+
+**Modularity:**  
+Compared to the previous approach of directly referencing TypeScript source files, this setup slightly increases modularity. Each package is now self-contained, with its compiled outputs and defined entry points in the `package.json`. By precompiling and packaging the library, it can be distributed outside the monorepo if needed, which enhances its modularity and reusability. However, the primary focus remains internal use within the monorepo.
+
+**Performance :**  
+This setup can slightly improve type-checking performance, especially within code editors. Since the type information is already generated as `.d.ts` files during precompilation, the editor can directly rely on these instead of processing TypeScript source files through project references. While cached project references can achieve similar speeds, precompiled declaration files might potentially reduce some overhead.
+
+## Which One Should I Choose?
+
+Here are some thoughts on which approach to use.
+
+**TypeScript Path Aliases: A Simple Option**
+
+TypeScript path aliases have been a reliable way to manage package resolution, particularly before package managers introduced the workspaces feature. They're straightforward to set up, requiring only a global `tsconfig.json` without additional. However, there are limitations to consider as they require additional bundling support/alias resolvers at runtime and might come with some performance degradation in large workspaces.
+
+{% callout type="deepdive" title="Isn't Nx using TS Path Aliases?" %}
+Yes and no. Nx has been around before package managers introduced workspaces. As a result, the default setup in Nx traditionally leveraged a root-level `tsconfig.base.json` containing path aliases to link packages within the monorepo.
```

Comment:
```suggestion
Yes and no. Nx has been around since before package managers introduced workspaces. As a result, the default setup in Nx traditionally leveraged a root-level `tsconfig.base.json` containing path aliases to link packages within the monorepo.
```